### PR TITLE
NFDIV-1620: update hub when final order is submitted

### PR DESCRIPTION
**Pull request by FaisalMoJ**
### Change description ###

Enter a description.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-1620

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/test/functional/features/final-order.feature [hunk 1]
```diff
@@ -25,6 +25,23 @@ Feature: Final order
     When I click "Continue"
     Then the page URL should be "/hub-page"
 
+    Given I set the case state to "FinalOrderRequested"
```
Comment (lara-mpt) [inline src/test/functional/features/final-order.feature:28]: isn't the case state already in FinalOrderRequested as a result of the lines above?

File: src/test/functional/features/final-order.feature [hunk 2]
```diff
@@ -25,6 +25,23 @@ Feature: Final order
...
+    And I click "Sign out"
+    When I login with applicant "1"
+    Then the page should include "You have applied for a ‘final order’. Your application will be checked by court staff."
+    And the page should include "You should receive an email within 2 working days,"
+    Given I click "Sign out"
+    When I login with applicant "2"
+    Then the page should include "Your wife has applied for a ‘final order’. The application will be checked by court staff."
```
Comment (lara-mpt) [inline src/test/functional/features/final-order.feature:35]: You could just do the check for app2 first and then sign out and check with app1, saves signing in and out twice

File: src/test/functional/features/final-order.feature [hunk 3]
```diff
@@ -25,6 +25,23 @@ Feature: Final order
...
+    And the page should include "You should receive an email within 2 working days,"
+    Given a superuser updates "dateFinalOrderNoLongerEligible" with "2020-01-01"
+    And I click "Sign out"
+    When I login with applicant "1"
+    Then the page should include "You will receive an email by"
+    Given I click "Sign out"
+    When I login with applicant "2"
+    And the page should include "You will receive an email by"
```
Comment (lara-mpt) [inline src/test/functional/features/final-order.feature:43]: same as above here

File: src/main/steps/applicant1/hub-page/sole/content.ts [hunk 4]
```diff
@@ -378,15 +381,17 @@ const cy: typeof en = (
     link: config.get('govukUrls.moneyAndProperty'),
   },
   finalOrderRequested: {
+    applicant2Line1: `Your ${partner} has applied for a ‘final order’.`,
+    applicant2Line2:
+      'A judge will review the application. You will then receive an email telling you what they decide.',
```
[the remark this answers is not in the document]
Comment (FaisalMoJ) [reply src/main/steps/applicant1/hub-page/sole/content.ts]: Renamed fields as suggested

File: src/test/functional/features/final-order.feature (hunk 2 above)
Comment (FaisalMoJ) [reply src/test/functional/features/final-order.feature]: Fixed the issue as suggested by Lara

File: src/test/functional/features/final-order.feature (hunk 1 above)
Comment (FaisalMoJ) [reply src/test/functional/features/final-order.feature]: Removed the line that sets state

File: src/test/functional/features/final-order.feature [hunk 5]
```diff
@@ -66,14 +68,22 @@ Feature: Final order
...
-    Given I go to "/"
     Then the page should include "Your wife has still not applied for a 'final order'"
-    Given I click "Apply for a final order"
-    Then the page URL should be '/respondent/finalising-your-application'
+    And the page URL should be '/respondent/hub-page'
 
-    Given I select "I want permission to apply for a final order, and to finalise my divorce"
+    Given a superuser updates "dateFinalOrderEligibleToRespondent" with "2020-01-01"
+    When I set the case state to "AwaitingFinalOrder"
+    And I click "Apply for a final order"
```
Comment (lara-mpt) [inline src/test/functional/features/final-order.feature:76]: Might be best to keep the page URL check after this step
```suggestion
    And I click "Apply for a final order"
    Then the page URL should be '/respondent/finalising-your-application'
```

**Review by lara-mpt — APPROVED** [review]

File: src/main/steps/applicant1/hub-page/sole/latest-update-content/final-order-requested.njk [hunk 6]
```diff
@@ -1,3 +1,8 @@
-{% for property, line in finalOrderRequested %}
-  <p class="govuk-body">{{ line }}</p>
-{% endfor %}
+{% if userCase.applicant1AppliedForFinalOrderFirst === 'Yes' %}
```
Comment (velimd) [inline src/main/steps/applicant1/hub-page/sole/latest-update-content/final-order-requested.njk:1]: Might be best to have this condition in the content file and pass it here that way you can also use the values from the definition file for 'Yes'

File: src/main/steps/respondent/hub-page/latest-update-content/final-order-requested.njk [hunk 7]
```diff
@@ -1,3 +1,8 @@
-{% for property, line in finalOrderRequested %}
-  <p class="govuk-body">{{ line }}</p>
-{% endfor %}
+{% if userCase.applicant2AppliedForFinalOrderFirst === 'Yes' %}
```
Comment (velimd) [inline src/main/steps/respondent/hub-page/latest-update-content/final-order-requested.njk:1]: same as mentioned above.

File: src/main/steps/respondent/hub-page/latest-update-content/final-order-requested.njk [hunk 8]
```diff
@@ -1,3 +1,8 @@
...
+    <p class="govuk-body">{{ finalOrderRequested.line2 }}</p>
+    <p class="govuk-body">{{ finalOrderRequested.line3 }}
+    <a href="https://www.gov.uk/get-help-with-court-fees"
```
Comment (velimd) [inline src/main/steps/respondent/hub-page/latest-update-content/final-order-requested.njk:4]: might be best to have the link and text in the content file as we do that in other places.

File: src/main/steps/respondent/hub-page/content.ts [hunk 9]
```diff
@@ -322,13 +326,18 @@ const cy: typeof en = ({ isDivorce, partner, userCase, contactEmail }: CommonCon
...
+    )} for the application before it can be submitted. Phone 0300 123 1711 to make payment. Have your card details ready.`,
+    line3:
+      'If you need help paying the fee then you will need to apply for Help With Fees first. Then phone with your Help With Fees reference' +
+      ' number. You can ',
+    line4: `${
       dayjs().isAfter(userCase.dateFinalOrderNoLongerEligible)
         ? `You will receive an email by ${getFormattedDate(
-            dayjs(userCase.dateFinalOrderSubmitted).add(config.get('dates.finalOrderSubmittedOffsetDays'), 'day'),
-            SupportedLanguages.Cy
+            dayjs(userCase.dateFinalOrderSubmitted).add(config.get('dates.finalOrderSubmittedOffsetDays'), 'day')
```
Comment (velimd) [inline src/main/steps/respondent/hub-page/content.ts:338]: if this is for the welsh property, worth adding the second argument (SupportedLanguages.Cy) in the function to get the Welsh text for the date format.

File: src/main/steps/respondent/hub-page/content.ts [hunk 10]
```diff
@@ -322,13 +326,18 @@ const cy: typeof en = ({ isDivorce, partner, userCase, contactEmail }: CommonCon
     line1: `Your ${partner} has applied for a ‘final order’. The application will be checked by court staff. If there are no other applications that need to be completed then your ${
       isDivorce ? 'divorce will be finalised' : 'civil partnership will be legally ended'
     }.`,
-    line2: `${
+    line2: `You need to pay ${getFee(
+      config.get('fees.finalOrderApplicationFee')
+    )} for the application before it can be submitted. Phone 0300 123 1711 to make payment. Have your card details ready.`,
+    line3:
+      'If you need help paying the fee then you will need to apply for Help With Fees first. Then phone with your Help With Fees reference' +
+      ' number. You can ',
```
Comment (velimd) [inline src/main/steps/respondent/hub-page/content.ts:334]: How comes this is broken up?

**Review by velimd — APPROVED** [review]

File: src/test/functional/features/final-order.feature (hunk 5 above)
Comment (FaisalMoJ) [reply src/test/functional/features/final-order.feature:76]: Changed as Lara suggested

File: src/main/steps/respondent/hub-page/latest-update-content/final-order-requested.njk (hunk 8 above)
Comment (FaisalMoJ) [reply src/main/steps/respondent/hub-page/latest-update-content/final-order-requested.njk]: Moved logic to content file as suggested

**Pull request merged**